### ISSUE TITLE
workspace blocked in cancelling state

### ISSUE THREAD

**Issue by etienne07C1** [coder/coder#7615]
Steps to reproduce:
I stopped a workspace, then restarted system before it had finished.
Upon coder restart, workspace state was still showing shutting down. 
I pressed cancel which blocked workspace and even after restarting coder, workspace remains in that state forever, 
making it unusable.

screenshot of the issue:
<img width="1426" alt="image" src="https://github.com/coder/coder/assets/1556719/a60763ae-8491-4ed3-a216-08576eef9dc7">

**Comment by sharkymark** [comment]
Not sure about what happened, but if delete does not work, use the CLI and `coder delete <workspace> --orphan and it will remove it from the database. If there is compute still running e.g., pod, container - you'll have to manually remove that.

**Comment by etienne07C1** [comment]
Thanks for reply. Does it mean I'll need to recreate workspace from template? 
Just to be sure my workspace is using docker volume resources I don't want to loose, will that operation keep them?

**Comment by matifali** [comment]
Deleting an orphan workspace by cli will remove it from the database only. Your volumes won't be deleted. 
The new workspace with from same template may or may not use the existing volumes depending how you name your volumes in the template. 

to be in the safe side, you should mount the volume to a container and backup your data then delete the volumes and create a fresh workspace.

**Comment by etienne07C1** [comment]
Yeah I tweaked the old `docker-build-image` template to mount my own volumes so that I'm always reusing same volumes across workspaces instead of creating new ones. 

I noticed the template I was using has been removed from coder examples so maybe outdated conf, might have caused my issue. Will close then

**Comment by etienne07C1** [comment]
@sharkymark  just to let you know I've tried deleting workspace but its state seems to forbid it:
<img width="342" alt="image" src="https://github.com/coder/coder/assets/1556719/1b61ce99-3722-45aa-9b6c-f9f2454e42b0">

any idea how I could force deleting it?
